include/swift/AST/DiagnosticsSema.def

```diff
@@ -3684,6 +3684,9 @@ ERROR(attr_not_on_variadic_parameters,none,
       "'%0' must not be used on variadic parameters", (StringRef))
 ERROR(attr_not_on_subscript_parameters,none,
       "'%0' must not be used on subscript parameters", (StringRef))
+ERROR(attr_not_on_enum_case_parameters,none,
+      "'%0' must not be used on enum case parameters", (StringRef))
```

Comment:
I believe spelling the rule based on what is allowed (rather than saying that something is not allowed here) is far more enlightening and likely to help the user.

Reply:
I agree, I think “can only be used on” is a better approach here because it has it carries a lot more useful information

Reply:
I agree with you two. I changed the error message to say something like `'inout' may only be used on function or initializer parameters`.